### PR TITLE
add random resource also for auto_login

### PR DESCRIPTION
**Pull request by davec82**
if you login with auto_login without a /resource,  resource for that jid will be 'null'

### CODE REVIEW

File: src/converse-core.js [hunk 1]
```diff
@@ -4893,7 +4893,12 @@
                         throw new Error("initConnection: If you use auto_login and "+
                             "authentication='login' then you also need to provide a password.");
                     }
-                    this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+Strophe.getResourceFromJid(this.jid);
+                    var resource = Strophe.getResourceFromJid(this.jid);
+                    if (!resource) {
+                        this.jid = this.jid.toLowerCase() + '/converse.js-' + Math.floor(Math.random()*139749825).toString();
+                    } else {
+                        this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+Strophe.getResourceFromJid(this.jid);
+                    }
```
Comment (jcbrand) [inline src/converse-core.js]: This line can be: `this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+resource;`

File: src/converse-core.js [hunk 2]
```diff
@@ -4893,7 +4893,12 @@
                         throw new Error("initConnection: If you use auto_login and "+
                             "authentication='login' then you also need to provide a password.");
                     }
-                    this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+Strophe.getResourceFromJid(this.jid);
+                    var resource = Strophe.getResourceFromJid(this.jid);
+                    if (!resource) {
+                        this.jid = this.jid.toLowerCase() + '/converse.js-' + Math.floor(Math.random()*139749825).toString();
```
Comment (jcbrand) [inline src/converse-core.js]: Can you please make a method `generateResource` (under the `Module-level functions` section) that generates the resource?

In other words, it contains the code: `/converse.js-' + Math.floor(Math.random()*139749825).toString();`

And then in the other place where the resource is generated that method should also be called.

Thanks

Comment (jcbrand) [inline src/converse-core.js]: Also please add an entry in the changelog in `docs/CHANGES.md`

**Comment by davec82** [comment]
Updated pull request, let me know if it's ok.

Thanks!

**Comment by jcbrand** [comment]
Looks good, thanks!

**Pull request merged**